src/main/scala/io/iohk/ethereum/domain/Blockchain.scala

```diff
@@ -187,8 +200,8 @@ class BlockchainImpl(
       into the case of having an incomplete best block and so an inconsistent db
      */
     val bestBlockNumberUpdates =
-      if (appStateStorage.getBestBlockNumber() > newBestBlockNumber)
-        appStateStorage.putBestBlockNumber(newBestBlockNumber)
+      if (appStateStorage.getBestBlockNumber() > potentialNewBestBlockNumber)
```

Comment:
do we want to keep this condition?
what if the `getBestBlockNumber` returns a number smaller than the potentialNewBestBlockNumber? we won't update it?

Reply:
As currently mantis is storing only the best chain, the guarantee is that we have a path from genesis up to the best block.

So the logic as I understand it is as follow: 
  - if we are deleting a block with a number lower or equal to the current best block, then we change the current best block because otherwise there would be a "hole" in the chain up to the current best block, so the best block can only be the parent of the block we are deleting.
  - if we are deleting a block with a number higher than what is considered the current best block, then it does not affect the current best block because otherwise it would mean extending the chain while we are just deleting a block (also we are not sure that we have a path to a random block above the best block number so this ensure consistency). I don't really know when this scenario happens to be honest, but I guess it might happen if we are deleting a series of block out of order.
  
So as of now I think the implementation still makes sense. But indeed it can not be used as is when we start storing more than the current best chain.